tests/PHPStan/Rules/Operators/data/invalid-binary.php

```diff
@@ -254,3 +254,14 @@ function benevolentPlus(array $a, int $i): void {
 		echo $k + $i;
 	}
 };
+
+function (int $int) {
+	$int + [];
+};
+
+function ($mixed, array $arr) {
+	if (is_array($mixed)) {
+		return;
+	}
+	$mixed + $arr; // mixed~array + array
```

Comment:
this line doesn't report an error, because of this early return

https://github.com/phpstan/phpstan-src/blob/3dffa1941cfe4bcaf7397dfc5aa54b2514564639/src/Rules/RuleLevelHelper.php#L169-L171

thats the actual reason for the changes discussed in https://github.com/phpstan/phpstan-src/pull/1579#discussion_r934784319

Reply:
Seems I need to double check this thesis, as thru rebasing things look different now.

I will be on vacation for a few days

Reply:
tried a few variants and came to 1355c5d2cfac6f2d70652cdf1a2efbc9f0b97a0b which made my local test script work, but breaks a lot of other stuff. I will take a break now, I can't wrap my head arround the current constellation

Reply:
extracted this case into a new issue https://github.com/phpstan/phpstan/issues/7863 - its not related to the actual issue we try to fix with this PR and can be done on its own